### DISCUSSION TITLE
A way to import schemas?

### DISCUSSION THREAD

**Discussion by ctron** [juhaku/utoipa#788] [Q&A]
The documented approach I see is:

```rust
#[derive(OpenApi)]
#[openapi(
    paths(),
    components(
        schemas(
            foo::Bar,
            foo::Baz,
        )
    ),
)]
pub struct ApiDoc;
```

Now, if I don't want to repeat myself, listing **every** type of the schema again, is there a way to import a full module? Like:

```rust
#[openapi(
    paths(),
    components(
        schemas(
            foo
        )
    ),
)]
```

**Comment by juhaku** [comment]
@ctron At the moment no, but not a bad idea. Earlier I was considering something similar where one could add macro on a module that would automatically collect schemas from it and then this module could be added to the `OpenApi` . But then I realized that creating another macro for this purpose and adding a macro on a module would make the code less explicit and more magical and would not serve the purpose of being obvious as possible.

**Comment by juhaku** [comment]
Created an issue about this: https://github.com/juhaku/utoipa/issues/989
